### PR TITLE
feat(redteam): add api healthcheck to redteam generate 

### PR DESCRIPTION
**Pull request by mldangelo**
### Summary

This pull request introduces centralized utilities for managing and checking the health of remote APIs. The changes include refactoring existing health check logic into a dedicated module and updating relevant parts of the codebase to utilize these new utilities.

### Key Modifications

- **New Module**: Added `apiHealth.ts` in the `util` directory to encapsulate API health check logic.
- **Refactored Logic**: Moved existing health check functions from `server.ts` to the new module.
- **Integration**: Updated `redteam` and `server` modules to use the centralized health check utilities.
- **Testing**: Added comprehensive unit tests for the new module and updated existing tests to reflect the changes.

### Implementation Details

- The `getRemoteHealthUrl` function determines the health check URL based on environment variables and cloud configuration.
- The `checkRemoteHealth` function performs the health check and returns a structured response.
- Updated tests ensure coverage for various scenarios, including error handling and configuration-based behavior.

### Testing Considerations

- Verified functionality with unit tests and integration tests.
- Mocked external dependencies to simulate various API responses.
- Ensured backward compatibility with existing configurations.

### Breaking Changes

No breaking changes are introduced in this update.

---

Please review the changes and provide feedback or approval for merging.

### CODE REVIEW

File: src/util/apiHealth.ts [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+    return customUrl.replace(/\/task$/, '/health');
+  }
+
+  const cloudConfig = new CloudConfig();
+  if (cloudConfig.isEnabled()) {
+    return `${cloudConfig.getApiHost()}/health`;
+  }
+
+  return 'https://api.promptfoo.app/health';
+}
```
Comment (sklein12) [inline src/util/apiHealth.ts]: this already exists in https://github.com/promptfoo/promptfoo/blob/main/src/redteam/remoteGeneration.ts

File: src/redteam/index.ts [hunk 2]
```diff
@@ -379,6 +381,22 @@ export async function synthesize({
...
+      const healthResult = await checkRemoteHealth(healthUrl);
+      if (healthResult.status !== 'OK') {
+        throw new Error(
+          `Unable to proceed with test generation: ${healthResult.message}\n` +
+            'Please check your API configuration or try again later.',
+        );
+      }
+      logger.debug('API health check passed');
+    }
+  }
```
Comment (sklein12) [inline src/redteam/index.ts:398]: If they're going to hit the unaligned provider we should check that they have access to that as well. There is a function to get that URL in https://github.com/promptfoo/promptfoo/blob/main/src/redteam/remoteGeneration.ts as well.

File: src/redteam/remoteGeneration.ts [hunk 3]
```diff
@@ -21,6 +21,28 @@ export function neverGenerateRemote(): boolean {
...
+ * @returns The health check URL, or null if remote generation is disabled.
+ */
+export function getRemoteHealthUrl(): string | null {
+  if (neverGenerateRemote()) {
+    return null;
+  }
+
+  const envUrl = getEnvString('PROMPTFOO_REMOTE_GENERATION_URL');
+  if (envUrl) {
+    return envUrl.replace(/\/task$/, '/health');
```
Comment (typpo) [inline src/redteam/remoteGeneration.ts]: is there a better way to do this?  seems a little brittle

File: src/util/apiHealth.ts [hunk 4]
```diff
@@ -0,0 +1,67 @@
...
+
+/**
+ * Checks the health of the remote API.
+ * @param url - The URL to check.
+ * @returns A promise that resolves to the health check response.
+ */
+export async function checkRemoteHealth(url: string): Promise<HealthResponse> {
+  try {
+    const cloudConfig = new CloudConfig();
+    const response = await fetchWithProxy(url, {
```
Comment (typpo) [inline src/util/apiHealth.ts]: sorry now that I see this, use `fetchWithTimeout` which calls `fetchWithProxy`.  Then you don't need your own timeout logic.

**Pull request merged**